File: PY/Bryann Valderrama/Matemática/numberOfDivisors.py
from sys import stdout


def divisors(x):
    nDiv = 1
    i = 2
    while i*i <= x:
        cnt = 0
        while x % i == 0:
            cnt += 1
            x //= i
        nDiv *= cnt + 1
        i += 1
    if x > 1:
        nDiv *= 2
    return nDiv


stdout.write(f'{divisors(100)}')

File: Java/Bryann Valderrama/Matemática/numberOfDivisors.java
public class Main {
    
    static int divisors(int x) {
        int nDiv = 1;
        for (int p = 2; p * p <= x; p++) {
            int cnt = 0;
            while (x % p == 0) {
                cnt++;
                x /= p;
            }
            nDiv *= cnt + 1;
        }
        if (x > 1) {
            nDiv *= 2;
        }
        return nDiv;
    }
    
    public static void main(String[] args) {
        System.out.println(divisors(100));
    }
}
